### ISSUE TITLE
oceDeleteData doesn't delete the accompanying "units" field. Do we care?

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#1565]
Was modifying an object by using `oceSetData()` to copy all the information from a given field to a new one with a different name (basically just a "renaming"). Then I used `oceDeleteData()` to get rid of the data field with the name I didn't like. However, I noticed that the `units` field for the deleted data item remained after the deletion.

Do we care? It seems sensible to delete any `units` fields associated with `@data` fields that are deleted (especially since `oceSetData()` will *create* a unit for anything that is added). Note that I don't think `oceDeleteMetadata()` would work for this purpose because it only affects the first level of `@metadata`.

On the other hand, anyone accessing the units should be doing so by name, i.e. `d[['units']]$oxygen` or better `d[['units']][['oxygen']]`. I don't have an example case where having units for fields that don't exist in `@data` would cause an issue, but I suppose there could be one.

**Comment by dankelley** [comment]
I agree that this remnant ought to be trimmed. I'll likely do it tomorrow.

**Comment by dankelley** [comment]
somehow I clicked to close, sorry

**Comment by dankelley** [comment]
I just fixed this. I also now remove flags.  It is only 4 lines of code, and this is such an isolated change that I feel comfortable in having done it in "develop" commit 3f259a62035a98c8c4d2f203c068fcb5c2861d18, prior to a CRAN release.  

The diff you get by clicking the commit link will show you that I've tested this using a station in the `section` dataset. 

I think this test is clear enough that we are ok on this update. Seem OK, @richardsc?
